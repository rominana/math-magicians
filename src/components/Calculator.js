import React from 'react';
import '../styles/Calculator.css';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = { value: 0 };
  }

  render() {
    const { value } = this.state;
    return (
      <div className="calculator">
        <div className="input">{value}</div>
        <div className="calculator-body">
          <button className="calculator-buttons" type="button">AC</button>
          <button className="calculator-buttons" type="button">+/&minus;</button>
          <button className="calculator-buttons" type="button">%</button>
          <button className="calculator-buttons operation" type="button">&divide;</button>
          <button className="calculator-buttons" type="button">7</button>
          <button className="calculator-buttons" type="button">8</button>
          <button className="calculator-buttons" type="button">9</button>
          <button className="calculator-buttons operation" type="button">&times;</button>
          <button className="calculator-buttons" type="button">4</button>
          <button className="calculator-buttons" type="button">5</button>
          <button className="calculator-buttons" type="button">6</button>
          <button className="calculator-buttons operation" type="button">&minus;</button>
          <button className="calculator-buttons" type="button">1</button>
          <button className="calculator-buttons" type="button">2</button>
          <button className="calculator-buttons" type="button">3</button>
          <button className="calculator-buttons operation" type="button">+</button>
          <button className="calculator-buttons cero-button" type="button">0</button>
          <button className="calculator-buttons" type="button">.</button>
          <button className="calculator-buttons operation" type="button">=</button>
        </div>
      </div>
    );
  }
}

export default Calculator;
