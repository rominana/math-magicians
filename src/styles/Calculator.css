* {
  margin: 0;
  padding: 0;
  background-color: black;
  box-sizing: border-box;
  font-size: 20px;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}

.section-title {
  color: white;
  font-size: 34px;
  padding: 4rem 4rem;
}

.calculator {
  min-width: 500px;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  margin-top: 50px;
}

.input {
  background-color: rgb(121, 121, 121);
  text-align: right;
  padding: 0.5rem;
  color: #fff;
  font-size: 3rem;
}

.calculator-body {
  display: grid;
  grid-template: repeat(5, 1fr) / repeat(4, 1fr);
}

.calculator-buttons {
  background-color: rgb(229, 229, 229);
  padding: 0.5rem;
  font-size: 1.5rem;
  border: 1px solid rgb(199, 199, 199);
  cursor: pointer;
}

.calculator-buttons:hover {
  background-color: rgb(204, 204, 204);
}

.operation {
  background-color: rgb(255, 125, 39);
}

.operation:hover {
  background-color: rgb(232, 115, 38);
}

.cero-button {
  grid-column: 1 / 3;
}
