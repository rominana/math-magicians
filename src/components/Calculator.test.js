/** @format */

import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from './Calculator';

describe('Calculator tests', () => {
  it('Renders Calculator component', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('simulate user interaction with the calculator: 3', () => {
    render(<Calculator />);
    const keyThree = screen.getByText('3');
    const input = document.querySelector('.input');
    fireEvent.click(keyThree);
    expect(input.textContent).toBe('3');
  });

  it('simulate the user interaction with the calculator: 7.4', () => {
    render(<Calculator />);
    const keySeven = screen.getByText('7');
    const keyDot = screen.getByText('.');
    const keyFour = screen.getByText('4');
    const input = document.querySelector('.input');
    fireEvent.click(keySeven);
    fireEvent.click(keyDot);
    fireEvent.click(keyFour);
    expect(input.textContent).toBe('7.4');
  });

  // pair programming test
  it('simulate the user interaction with the calculator: 10', () => {
    render(<Calculator />);
    const keyFive = screen.getByText('5');
    const keyTimes = screen.getByText('x');
    const keyTwo = screen.getByText('2');
    const keyEqual = screen.getByText('=');
    const input = document.querySelector('.input');
    fireEvent.click(keyFive);
    fireEvent.click(keyTimes);
    fireEvent.click(keyTwo);
    fireEvent.click(keyEqual);
    expect(input.textContent).toBe('10');
  });
});
