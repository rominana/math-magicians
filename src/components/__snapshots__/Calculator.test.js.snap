// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator tests Renders Calculator component 1`] = `
<div
  className="calculator-container"
>
  <h2
    className="section-title"
  >
    Let's do some math!
  </h2>
  <div
    className="calculator"
  >
    <div
      className="input"
    >
      0
    </div>
    <div
      className="calculator-body"
    >
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="calculator-buttons operation"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="calculator-buttons operation"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="calculator-buttons operation"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="calculator-buttons operation"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="calculator-buttons cero-button"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="calculator-buttons operation"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Calculator tests Renders Calculator component correctly 1`] = `
<div
  className="calculator-container"
>
  <h2
    className="section-title"
  >
    Let's do some math!
  </h2>
  <div
    className="calculator"
  >
    <div
      className="input"
    >
      0
    </div>
    <div
      className="calculator-body"
    >
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="calculator-buttons operation"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="calculator-buttons operation"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="calculator-buttons operation"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="calculator-buttons operation"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="calculator-buttons cero-button"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="calculator-buttons"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="calculator-buttons operation"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
